zulipterminal/ui_tools/views.py

```diff
@@ -934,36 +934,54 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
 PopUpViewTableContent = Sequence[Tuple[str, Sequence[Union[str, Tuple[str, str]]]]]
 
 
-class PopUpView(urwid.ListBox):
+class PopUpView(urwid.Frame):
     def __init__(
         self,
         controller: Any,
-        widgets: List[Any],
+        body: List[Any],
         command: str,
         requested_width: int,
         title: str,
+        header: Optional[List[Any]] = None,
+        footer: Optional[List[Any]] = None,
     ) -> None:
         self.controller = controller
         self.command = command
         self.title = title
-        self.log = urwid.SimpleFocusListWalker(widgets)
+        self.log = urwid.ListBox(urwid.SimpleFocusListWalker(body))
```

Comment:
Do you mean we change it to something like `self.body`?

Reply:
Ya that sounds better.

Reply:
Another thing, I know I can access the log using `self.body._w` or something like that but it would be easier if I could just access `self.log` if I wanted to manipulate the log? So you could keep the original line and add the listbox line under?

Reply:
Do you mean something like this?
```python3
self.log = urwid.SimpleFocusListWalker(body)
body = urwid.ListBox(self.log)
```

Reply:
Also, I don't think `ListBox` provides a `._w` attribute. We could however do `self.log.body`(no name change) to get the `SimpleFocusListWalker` object containing the list of widgets. The [urwid docs](http://urwid.org/reference/list_walkers.html#urwid.SimpleFocusListWalker) states that:

> Changes made to this object (when it is treated as a list) are detected automatically and will cause ListBox objects using this list walker to be updated.

But I don't get how are we supposed to update the list walker body.

Reply:
Ok, got it. The ListWalker object is like a `list`. So you could simply access/modify the widgets using their index, IG

Reply:
Yes we could do that, but this is simpler to understand right?
```
self.log = urwid.SimpleFocusListWalker(body)
body = urwid.ListBox(self.log)
```
body would indicate body of Frame and self.log can be used to access the list walker

Reply:
Sure, it looks good. I'll do the change.